File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Inheritance/MultipleSubclassesNoOverrides.swift
@main
struct MyMain {

    static func main() {
        let someVehicle = Vehicle()
        let bicycle = Bicycle()
        bicycle.hasBasket = true
        let tandem = Tandem()
        tandem.hasBasket = true
        tandem.currentNumberOfPassengers = 2
        tandem.currentSpeed = 22.0
        let des = tandem.description
    }
}
class Vehicle {
    var currentSpeed = 0.0
    var description: String {
        return "traveling at \(currentSpeed) miles per hour"
    }
    func makeNoise() {
        // do nothing - an arbitrary vehicle doesn't necessarily make a noise
    }
}

class Bicycle: Vehicle {
    var hasBasket = false
}

class Tandem: Bicycle {
    var currentNumberOfPassengers = 0
}

File: Swift-C++-Testsuite/test/structsAndClasses/C++/Classes/MultipleSubclassesNoOverrides.cpp

#include <iostream>

using namespace std;

class Vehicle {

public:
  double currentSpeed = 0.0;
  string getDescription() {
    return "traveling at " + to_string(currentSpeed) + " miles per hour";
  };
  virtual void makeNoise() {
    // do nothing - an arbitrary vehicle doesn't necessarily make a noise
  }
};

class Bicycle: public Vehicle {
public:
  bool hasBasket = false;
};

class Tandem: public Bicycle {
public:
  int currentNumberOfPassengers = 0;
};

int main() {
  Vehicle *someVehicle = new Vehicle();
  Bicycle *bicycle = new Bicycle();
  bicycle->hasBasket = true;
  Tandem *tandem = new Tandem();
  tandem->hasBasket = true;
  tandem->currentNumberOfPassengers = 2;
  tandem->currentSpeed = 22.0;
  string des = tandem->getDescription();

  delete someVehicle;
  delete bicycle;
  delete tandem;

  return 0;
}